### ISSUE TITLE
`0.4.1` - Kubeconfig resource times out

### ISSUE THREAD

**Issue by muhlba91** [pulumiverse/pulumi-talos#112]
When using the `talos.cluster.Kubeconfig` or `talos.cluster.getKubeconfigOutput` resource, the provider is not able to retrieve a working kubeconfig.
Before `0.4.1` I was on `0.2.0` and used `talos.cluster.getKubeconfigOutput` with the same parameters. This worked flawlessly before the update.
It just waits and times out now.

The code looks like:

```typescript
new talos.cluster.Kubeconfig('talos-kubeconfig',
    {
          clientConfiguration: talosSecrets.clientConfiguration,
          node: talosConfig.machine.network.ip.v4,
    },
    {}
)
```

The error is (this goes on until Pulumi times out the resource creation):

```
    2024/11/06 07:59:20 [DEBUG] Waiting for state to become: [success]
    2024/11/06 07:59:20 [TRACE] Waiting 500ms before next try
    2024/11/06 07:59:20 [TRACE] Waiting 1s before next try
    2024/11/06 07:59:21 [TRACE] Waiting 2s before next try
    2024/11/06 07:59:23 [TRACE] Waiting 4s before next try
    2024/11/06 07:59:27 [TRACE] Waiting 8s before next try
```

**Comment by shkarface** [comment]
I have the same problem as well, any idea @ringods ?

**Comment by xzizka** [comment]
Could you please try to set the env variable `export GRPC_ENFORCE_ALPN_ENABLED=false` before you run the `pulumi up`?
See more here: https://github.com/pulumiverse/pulumi-talos/issues/113

**Comment by muhlba91** [comment]
thank you, this works now.

**Comment by shkarface** [comment]
I can also confirm that the new `go-grpc` was the problem, setting the env var or upgrading to talos 1.8.3 fixed the issue.

**Comment by xzizka** [comment]
> I can also confirm that the new `go-grpc` was the problem, setting the env var or upgrading to talos 1.8.3 fixed the issue.

@shkarface So if I use Talos 1.8.3 as an image, it does not need the env variable mentioned above?

**Comment by shkarface** [comment]
> > I can also confirm that the new `go-grpc` was the problem, setting the env var or upgrading to talos 1.8.3 fixed the issue.
> 
> 
> 
> @shkarface So if I use Talos 1.8.3 as an image, it does not need the env variable mentioned above?

No, as Talos 1.8.3 already contains the ALPN in the TLS extensions response

**Comment by ringods** [comment]
Sorry everyone, but I haven't had the time to follow up on the Talos provider issues. Is this problem still manifesting itself with the latest releases?

**Comment by xzizka** [comment]
> Sorry everyone, but I haven't had the time to follow up on the Talos provider issues. Is this problem still manifesting itself with the latest releases?

Hello @ringods . With version 1.8.3 the issue has been solved. Thanks.
